### PR TITLE
Improve Compatibily with macOS Contacts

### PR DESCRIPTION
**Pull request by zlajo**
I would like to use nextcloud contacts as a replacement to storing my contacts in iCloud. Unfortunately there are some things that don't work as expected and while some issues require fundamental changes (like [#609](https://github.com/nextcloud/contacts/issues/609)) there are some which I was able to fix with only a few small changes.

One thing I found out was that there already is support for hiding contact groups (as defined by [https://tools.ietf.org/html/rfc6350#section-6.1.4](https://tools.ietf.org/html/rfc6350#section-6.1.4)). macOS Contacts still uses vCard 3.0 and because of that cannot relay on the vCard kind property defined by vCard 4.0. But there is a proprietary attribute which does the same thing (X-ADDRESSBOOKSERVER-KIND). I updated the contacts app to also respect this Apple specific property and added an easy way to add further proprietary "kind"-properties. 

There is also a problem (at least I think so) with the underlying vCard serialization which leads to strange property names. See my comment in [src/store/contacts.js](https://github.com/zlajo/contacts/blob/e87629a46721a54c1ac50d7441ff87673e5a961d/src/store/contacts.js#L30) for a detailed description of the problem and a potential workaround. 

Please let me know if my update works and contact me if further changes/discussions are required.

### CODE REVIEW

File: src/models/contact.js
Comment (skjnldsv) [inline src/models/contact.js:49]: I would prefer this as module export constants please :)

File: src/store/addressbooks.js
Comment (skjnldsv) [inline src/store/addressbooks.js:349]: Would be much cleaner! Nice! :+1:

File: src/store/contacts.js
Comment (skjnldsv) [inline src/store/contacts.js:45]: Nice! I was not aware of such possibilities :)

**Review by skjnldsv — CHANGES_REQUESTED** [review]
See comments :)

File: src/store/contacts.js
Comment (skjnldsv) [reply src/store/contacts.js:45]: Ah, yes, it was on the old designSet update workaround! Nice! :+1:

**Comment by zlajo** [comment]
@skjnldsv I have just updated my pull request.

File: src/models/contact.js
Comment (skjnldsv) [inline src/models/contact.js:49]:
```suggestion
export const ContactKindProperties = ['KIND', 'X-ADDRESSBOOKSERVER-KIND']

export const MinimalContactProperties = [
	'EMAIL', 'UID', 'CATEGORIES', 'FN', 'ORG', 'N', 'X-PHONETIC-FIRST-NAME', 'X-PHONETIC-LAST-NAME',
].concat(ContactKindProperties)
```

**Review by skjnldsv — COMMENTED** [review]
One last cleanup on the code :+1: 
Looks great! Nice job :)

**Review by skjnldsv — APPROVED** [review]
Good for me :)
Please rebase and squash your commits into a single one before merging! :rocket:

**Comment by jancborchardt** [comment]
@zlajo only the Developer Code of Origin test fails, could you sign off your commit? :) You just have to `git commit --amend --signoff` – more info at https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

**Comment by jancborchardt** [comment]
By the way @zlajo we have a Nextcloud Talk instance for easier communication – we also have a channel for Groupware/Contacts there. If you like, we can invite you? :)

**Comment by zlajo** [comment]
@jancborchardt I just did a rebase on the current master and a force push. Is this what I had to do?

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**
